src/main/java/com/iexec/worker/replicate/ReplicateDemandService.java

```diff
@@ -67,11 +68,12 @@ public ReplicateDemandService(IexecHubService iexecHubService,
      * We use single thread executor to make sure the worker does not ask for more
      * than one replicate at the same time. The executor's queue is of size 1 to
      * avoid memory leak if the thread halts for any reason.
+     * @return A {@link CompletableFuture} representing the executing task.
      */
-    @Scheduled(fixedRateString = "#{publicConfigurationService.askForReplicatePeriod}")
-    void triggerAskForReplicate() {
+    @Scheduled(fixedDelayString = "#{publicConfigurationService.askForReplicatePeriod}")
```

Comment:
I agree, this makes more sense. As a side effect (positive one I think) this would also add some randomness between workers due to I/O and network differences. It is minor but could be interesting at scale.